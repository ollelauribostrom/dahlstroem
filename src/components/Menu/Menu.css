.menu {
}

.menu-item__wrapper {
  position: relative;
  margin-left: 30px;
}

.menu-item__link {
  text-decoration: none;
}

@media only screen and (min-device-width: 2000px) {
  .menu-item__link {
    font-size: 50px;
  }
}
