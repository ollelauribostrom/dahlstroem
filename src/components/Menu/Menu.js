import {Link, StaticQuery, graphql} from 'gatsby';
import PropTypes from 'prop-types';
import React from 'react';

import DropDown from '../DropDown';

const MenuItem = props => (
  <span
    tabIndex={0}
    style={{
      color: props.color.hex,
      cursor: 'pointer',
      borderBottom: props.isActive ? `1px solid ${props.color.hex}` : 'none',
    }}
    {...props}
  >
    {props.children}
  </span>
);

class Menu extends React.Component {
  constructor (props) {
    super (props);
    this.state = {showDropDown: false};
  }
  render () {
    console.log (this.props.path);
    return (
      <StaticQuery
        query={graphql`
          query {
            allDatoCmsCategory {
              edges {
                node {
                  name
                  slug
                }
              }
            }
          }
        `}
        render={data => {
          return (
            <div
              style={{
                display: 'flex',
                alignItems: 'center',
              }}
            >
              <div
                style={{position: 'relative', marginLeft: '20px'}}
                onMouseLeave={() => this.setState ({showWorkDropDown: false})}
              >
                <MenuItem
                  isActive={false}
                  onMouseOver={() => this.setState ({showWorkDropDown: true})}
                  onFocus={() => this.setState ({showWorkDropDown: true})}
                  color={this.props.color}
                >
                  Work
                </MenuItem>
                {this.state.showWorkDropDown
                  ? <DropDown
                      categories={data.allDatoCmsCategory.edges.map (edge => ({
                        title: edge.node.name,
                        href: `/${edge.node.slug}/`,
                      }))}
                      onBlur={() => this.setState ({showWorkDropDown: false})}
                      color={this.props.menuColor}
                      background={this.props.menuBackground}
                    />
                  : null}
              </div>
              <div style={{position: 'relative', marginLeft: '20px'}}>
                <MenuItem
                  isActive={this.props.path === '/resume'}
                  onMouseOver={() => this.setState ({showAboutDropDown: true})}
                  color={this.props.color}
                >
                  <Link
                    to="/resume"
                    style={{
                      color: this.props.color.hex,
                      textDecoration: 'none',
                    }}
                  >
                    Resume
                  </Link>
                </MenuItem>
              </div>
              <div style={{position: 'relative', marginLeft: '20px'}}>
                <MenuItem
                  isActive={this.props.path === '/contact'}
                  onMouseOver={() => this.setState ({showAboutDropDown: true})}
                  color={this.props.color}
                >
                  <Link
                    to="/contact"
                    style={{
                      color: this.props.color.hex,
                      textDecoration: 'none',
                    }}
                  >
                    Contact
                  </Link>
                </MenuItem>
              </div>
            </div>
          );
        }}
      />
    );
  }
}

Menu.propTypes = {};

export default Menu;
