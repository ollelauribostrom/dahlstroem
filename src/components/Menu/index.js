import Menu from './Menu';
import './Menu.css';

export default Menu;
