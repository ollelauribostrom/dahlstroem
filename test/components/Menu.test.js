import React from 'react';
import { shallow } from 'enzyme';

import Menu from '../../src/components/Menu';

describe('Tests for Menu', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Menu
        location={{ pathname: '/test' }}
        color={{ hex: '#fff' }}
        menuColor={{ hex: '#fff' }}
        menuBackground={{ hex: '#000' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
