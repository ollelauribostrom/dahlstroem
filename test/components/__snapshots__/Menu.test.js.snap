// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Menu renders correctly 1`] = `
<div
  className="menu"
>
  <MenuItem
    isActive={false}
    title="Work"
    url="/"
  />
  <MenuItem
    isActive={false}
    title="Resume"
    url="/resume"
  />
  <MenuItem
    isActive={false}
    title="Contact"
    url="/contact"
  />
</div>
`;
